.github/workflows/build-images.yaml

```diff
@@ -107,12 +107,17 @@ jobs:
         if: ${{ github.ref_name == env.MAIN_BRANCH_NAME }}
         id: add-latest-tag
         run: |
+          echo "TAG_NAME=latest" >> $GITHUB_ENV
           echo "IMG_TAGS=latest ${{ env.IMG_TAGS }}" >> $GITHUB_ENV
-      - name: Add release tag
+      - name: Add tag based on branch
         if: ${{ github.ref_name != env.MAIN_BRANCH_NAME }}
         id: add-branch-tag
         run: |
+          if [[ ${GITHUB_REF_NAME/\//-} =~ ^v[0-9]+\.[0-9]+\.[0-9]+(-.+)?$ ]]; then
```

Comment:
What about preventing all this ugly conditionals with a new `build-image-on-tag.yaml` with 
```yaml
name: Publish in GitHub Package Registry on tag push

on:  
  push:
    tags:
      - 'v*'
```

Reply:
I reckon that in order to get rid of all the conditionals we might need either specific steps or workflows to the following options for not only versions and tags, but dependencies too:

1. Push: main, pull request regular branch, release and sha
2. Scheduled: sha builds 
3. Workflow dispatch: custom

Might be good to address the cleaner way including all our use cases in a following PR?